app/consumer/genesis_test.go

```diff
@@ -170,52 +424,247 @@ func getGenesisTransformCmd() (*cobra.Command, error) {
 	return cmd, err
 }
 
+// transformConsumerGenesis transform json content in a given file and return
+// the consumer genesis transformation result or an error
+// - filePath    file with consumer genesis content in json format
+// - version     ICS version to which the data should be transformed to
+func transformConsumerGenesis(filePath string, version *string) ([]byte, error) {
+	cmd, err := getGenesisTransformCmd()
+	if err != nil {
+		return nil, fmt.Errorf("Error setting up transformation command: %s", err)
+	}
+
+	args := []string{}
+	if version != nil {
+		args = append(args, fmt.Sprintf("--to=%s", *version))
+	}
+	args = append(args, filePath)
+
+	cmd.SetArgs(args)
+
+	result := new(bytes.Buffer)
+	cmd.SetOutput(result)
+	_, err = cmd.ExecuteC()
+	if err != nil {
+		return nil, fmt.Errorf("Error runnning transformation command: %v", err)
+	}
+	return result.Bytes(), nil
+}
+
 // Check transformation of a version 2 ConsumerGenesis export to
 // consumer genesis json format used by current consumer implementation.
-func TestConsumerGenesisTransformationV2(t *testing.T) {
-	version := "v2"
-	filePath := filepath.Join(t.TempDir(), "oldConsumerGenesis.json")
+func TestConsumerGenesisTransformationFromV2ToCurrent(t *testing.T) {
+	version := "v2.x"
+	ctx := getClientCtx()
 
-	err := os.WriteFile(
-		filePath,
-		[]byte(consumerGenesisStates[version]),
-		fs.FileMode(0o644))
+	srcGenesis := consumerTypes.GenesisState{}
+	err := ctx.Codec.UnmarshalJSON([]byte(consumerGenesisStates[version]), &srcGenesis)
+	require.NoError(t, err, "Error parsing old version of ccv genesis content for consumer")
+
+	filePath := createConsumerDataGenesisFile(t, version)
+	defer os.Remove(filePath)
+	resultGenesis := consumerTypes.GenesisState{}
+	result, err := transformConsumerGenesis(filePath, nil)
+	require.NoError(t, err)
+	err = ctx.Codec.UnmarshalJSON(result, &resultGenesis)
 	require.NoError(t, err)
+
+	// Some basic sanity checks on the content.
+	require.NotNil(t, resultGenesis.Provider.ClientState)
```

Comment:
It feels a bit brittle to me, since it's very hard to see that this actually makes the v2 genesis into a genesis for the current version - is there a way you could parse the transformed genesis similar to how it actually would be parsed?

Reply:
agree that this is hard to figure out as one would expect v2 types being use here as target version.
pls note that this is due to the problem not being able to import the v2 types due to module conflicts... which would be needed to properly test this.
also for encoding reasons I can't just create custom generic target structs faking v2 - which would also be brittle/tradeoff

Reply:
Ok, I expected that and I think the solution you came up with is exactly a good tradeoff :)